### ISSUE TITLE
shorthand_operator is disabled

### ISSUE THREAD

**Issue by sk-** [SwifterSwift/SwifterSwift#916]
> Please fill out this template when filing an issue.
>
> All comment lines beginning with an **>** instruct you with what info we expect. You can delete those lines once you've filled in the info.
>

[x] I've read, understood, and done my best to follow the [Contributing guidelines](https://github.com/SwifterSwift/SwifterSwift/tree/master/CONTRIBUTING.md) before opening this issue.

### What did you do?
I'm evaluating replacing some of our extensions with SwifterSwift, which overall looks really nice and complete, however I found that `shorthand_operator` is disabled.

The rationale for that check is to prevent creating temporary objects when using the inline operators.

So instead of implementing `+=` based on `+`, one should prefer to implement `+` based on `+=`, like:

```
public func + (left: A, right: A) -> A {
  var result = left
  result += right
  return result
}
```

Is there a technical rationale for the current implementation, if not, would you open to accept a PR changing the implementation? There are only 3 files affected, see https://github.com/SwifterSwift/SwifterSwift/search?q=shorthand_operator.

### What did you expect to happen?
> Please replace this with what you expected to happen.

### What happened instead?
> Please replace this with of what happened instead.

### SwifterSwift Environment

- SwifterSwift version:
- Xcode version:
- macOS version running Xcode:
- Swift version:
- Platform(s) running SwifterSwift:

### Demo Project
>Please link to or upload a project we can download that reproduces the issue.

**Comment by LucianoPAlmeida** [comment]
Hi @sk- , can you elaborate on why this is better? 

> The rationale for that check is to prevent creating temporary objects when using the inline operators.

What is exactly the "temporary object" in this situation? Can you elaborate on why this is specific for inline?

**Comment by sk-** [comment]
@LucianoPAlmeida in the current inline operators implementations the following is happening:

```swift
static func += (lhs: inout CGPoint, rhs: CGPoint) {
    // swiftlint:disable:next shorthand_operator
    lhs = lhs + rhs
}
```

1) Allocates and initializes a new instance (inside `+`)
2) Then performs some mutation to prepare that new instance (inside `+`)
3) Sets `lhs` to point to the new instance
4) Releases the old `lhs`, possibly deallocating it

See https://github.com/realm/SwiftLint/issues/1828#issuecomment-328230004 for more details.

Whereas, if you implement it the other way around, you only have step 2 for inline operations.

**Comment by LucianoPAlmeida** [comment]
Ah I see the point, I think I got confused by the `inline` terminology, but yeah this seems a legit change to be made, feel free to open the PR :)

**Comment by sk-** [comment]
You are right, sorry for the confusion. I meant `in-place` instead of inline.

**Comment by LucianoPAlmeida** [comment]
Fixed by #932, I just cherry-picked the relevant form your changes @sk-, so thanks :)
